content/en/blog/_posts/2021-04-08-introducing-indexed-jobs.md

```diff
@@ -0,0 +1,89 @@
+---
+title: "Introducing Indexed Jobs"
+date: 2021-05-05
+slug: introducing-indexed-jobs
+---
+
+**Author:** Aldo Culquicondor (Google)
+
+Once you have containerized a non-parallel {{< glossary_tooltip text="Job" term_id="job" >}},
+it is quite easy to get it up and running on Kubernetes without modifications to
+the binary. But when running parallel distributed Jobs, in most cases, you had
+to set a separate system to partition the work among the workers. For
+example, you could set up a task queue to [assign one work item to each
+Pod](/docs/tasks/job/coarse-parallel-processing-work-queue/) or [multiple items
+to each Pod until the queue is emptied](/docs/tasks/job/fine-parallel-processing-work-queue/).
+
+SIG Apps introduces the Job _completion mode_, which allows you to control how
+Pod completions affect the overall progress of a Job and, with 2 possible values
+(for now):
+
+- `NonIndexed` (default): the Job is considered complete when there have been
+  a number of successfully completed Pods equal to the specified number in
+  `.spec.completions`. In other words, each Pod completion is homologous to
+  each other. Any Job you might have created before the introduction of
+  completion modes is implicitly NonIndexed.
+- `Indexed`: the Job is considered complete when there is one successfully
+  completed Pod associated to each index from 0 to `.spec.completions-1`. Each
+  Pod gets its associated index in the annotation
+  `batch.kubernetes.io/job-completion-index`.
+
+You can start using Jobs with Indexed completion mode, or Indexed Jobs, for
+short, to easily start parallel Jobs in which each worker works on a partition
+of the data that is statically assigned. This saves you from having to set up
+a queuing system or even having to modify your binary!
+
+## Creating an Indexed Job
+
+To create an Indexed Job, you just have to add `completionMode: Indexed` to the
+Job spec and make use of the `JOB_COMPLETION_INDEX` environment variable.
+
+```yaml
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: 'sample-job'
+spec:
+  completions: 3
+  parallelism: 3
+  completionMode: Indexed
+  template:
+    spec:
+      restartPolicy: Never
+      containers:
+      - command:
+        - 'bash'
+        - '-c'
+        - 'echo "My partition: ${JOB_COMPLETION_INDEX}"'
+        image: 'docker.io/library/bash'
+        name: 'sample-load'
+```
+
+When you run the example, you will see that each of the 3 created Pods gets a
+different completion index. For users convenience, the control plane sets the
+`JOB_COMPLETION_INDEX` environment variable, but you can choose to [set your
+own](/docs/tasks/inject-data-application/environment-variable-expose-pod-information/)
+or [expose the index as a file](/downward-api-volume-expose-pod-information/).
```

Comment:
In the Netlify preview the /downward-api-volume-expose-pod-information/ URL results in a 404 error. I believe the intended target is https://v1-17.docs.kubernetes.io/docs/tasks/inject-data-application/downward-api-volume-expose-pod-information/ and requires the complete path "/docs/tasks/inject-data-application/downward-api-volume-expose-pod-information/" to build properly.

Reply:
Oops, copy-and-paste mistake.